homeworks/veizerattila/2_vars_datatypes/user_info.py

```diff
@@ -0,0 +1,120 @@
+prog_skills = input("Give me 4 programming languages you're familiar with, separated by a comma! ")
+
+if ( prog_skills.count(",") + 1 ) == 4:
+# Az if igaz ága csak akkor enged át, ha valaki 4 inputot adott meg, 3 vesszővel elválasztva
+
+    skills = {"skills": prog_skills.split(",")}
+    user_info.update(skills)
+
+    pp.pprint(user_info)
+    print("==========================================================================================================")
+
+
+    # 2. Rendezd a favourite_meals lista elemeit abc szerinti növekvő sorrendbe.
+    keys = list(user_info.keys()) # kulcs változó definiálása, hogy így a dict. elemeire a key alapján k;s[bbi lehessen hivatkozni (pl.: "favourite_meals" helyett "2")
+    user_info[keys[2]] = sorted(user_info[keys[2]])
+    print("2. Rendezd a favourite_meals lista elemeit abc szerinti növekvő sorrendbe!")
+    pp.pprint(user_info)
+    print("==========================================================================================================")
+
+
+    # 3. Printeld ki a favourite_meals lista utolsó előtti elemét
+    print("3. Printeld ki a favourite_meals lista utolsó előtti elemét!")
+    pp.pprint(f"A 'favourite_meals' lista utolsó előtti eleme: {user_info[keys[2]][-2]}")
+    print("==========================================================================================================")
+
+
+    # 4. Adj hozzá egy “spaghetti” string-et ugyanehhez a listához.
+    user_info[keys[2]].append("spaghetti")
+    print("4. Adj hozzá egy “spaghetti” string-et ugyanehhez a listához!")
+    pp.pprint(user_info)
+    print("==========================================================================================================")
+
+
+    # 5. Add hozzá a favourite_meals-hez az aktuális favourit_meals lista harmadik és negyedik elemét (nem az index-ét) újra.
+    user_info[keys[2]].extend(["sushi" , "spaghetti"])
+    print("5. Add hozzá a favourite_meals-hez az aktuális favourit_meals lista harmadik és negyedik elemét (nem az index-ét) újra!")
+    print("megj.: a listának most nincs 4. eleme, a legutolsó elem a 3. index (most: 'spaghetti')")
+    pp.pprint(user_info)
+    print("==========================================================================================================")
+
+
+    # 6. Ezután töröld az így keletkezett duplikátumokat!
+    user_info[keys[2]] = sorted(list(set(user_info[keys[2]]))) # itt újrarendeztem abc szerint
+    print("6. Ezután töröld az így keletkezett duplikátumokat!")
+    pp.pprint(user_info)
+    print("==========================================================================================================")
+
+
+    # 7. Cseréld fel a favourite_meals lista első és utolsó elemét! (Megj.: itt sem indexet használtam, hanem sima sorrendet.)
+    ind1 , ind2 = 0 , -1
+    user_info[keys[2]][0] , user_info[keys[2]][-1] = user_info[keys[2]][ind2] , user_info[keys[2]][ind1]
```

Comment:
tök jó a logikád, de szerintem hosszabb leírni az index helyére hogy ind2 és ind1 mintha csak beírnád hogy -1 és 0 :)

Reply:
Köszi, talán kicsit túlgondoltam :)